app/ui/src/main/java/com/fsck/k9/activity/compose/AttachmentPresenter.java

```diff
@@ -110,16 +120,45 @@ private boolean hasLoadingAttachments() {
         return false;
     }
 
-    public ArrayList<Attachment> createAttachmentList() {
+    public ArrayList<Attachment> createAttachmentListWithoutResizing() {
         ArrayList<Attachment> result = new ArrayList<>();
         for (Attachment attachment : attachments.values()) {
             result.add(attachment);
         }
         return result;
     }
 
-    public List<com.fsck.k9.message.Attachment> getAttachments() {
-        return new ArrayList<com.fsck.k9.message.Attachment>(attachments.values());
+    public ArrayList<Attachment> createAttachmentList() {
+        ArrayList<Attachment> result = new ArrayList<>();
+        for (Attachment attachment : attachments.values()) {
+            int resizeCircumference;
+            int resizeQuality;
+            if (account.isResizeImageEnabled() && !attachment.resizeImageOverrideDefault && ImageResizer.isImage(context, attachment.uri)) {
+                resizeCircumference = account.getResizeImageCircumference();
+                resizeQuality = account.getResizeImageQuality();
+            } else if (attachment.resizeImageOverrideDefault && ImageResizer.isImage(context, attachment.uri)) {
+                resizeCircumference = attachment.resizeImageCircumference;
+                resizeQuality = attachment.resizeImageQuality;
+            } else {
+                resizeCircumference = 0;
+                resizeQuality = 0;
+            }
+
+            if (resizeCircumference != 0) {
+                try {
+                    String newFilename = imageResizer.getResizedImageFile(context, attachment.uri, resizeCircumference, resizeQuality);
+                    long size = (new File(newFilename)).length();
```

Comment:
Either make `ImageResizer.getResizedImageFile()` return a file path and file size or a `File` instance if you need one right afterwards anyway.